pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -207,6 +207,93 @@ def restore_training_state(self, checkpoint, load_optimizer_states: bool = True)
         for scheduler, lrs_state in zip(self.trainer.lr_schedulers, lr_schedulers):
             scheduler['scheduler'].load_state_dict(lrs_state)
 
+    def restore_callbacks(self) -> None:
+        """ Restores all callbacks from the pre-loaded checkpoint. """
+        if not self._loaded_checkpoint:
+            return
```

Comment:
is there a risk of these restoration functions being called outside this context? should the start and end restore from checkpoint be replaced with a dedicated context manager for resuming?

Reply:
Hey, good question. 
If you look at the "end result" #7652 (open to discussion) you will see here in the Trainer file resume_start() and resume_end are actually in very different places(), so I can't make it into a context manager.

Yes, I think it's best to document the order. The order may be important. In the future, we will want to enable configuration of what is restored, so some of these functions get called on demand and some won't be called.

Reply:
actually, maybe a context manager could still work. I will investigate it in #7652

Reply:
I think Ananth's suggestion is good.

Also, could any other class want to call the `start` and `end` methods?

Reply:
No, I think we would only want to call it for unit testing, or the context manager if that works. 
So I know what you are saying, yes I will put the underscores everywhere

Reply:
okay, ctx manager could kinda work but I see an issue. can we move the conversation to #7652 so I can directly point to the code in trainer.py?